internal/cmd/root.go

```diff
@@ -65,8 +65,14 @@ func preRunCheck(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
+	cmdName := cmd.Name()
+	if cmd.Parent() != cmd.Root() {
+		// e.g. `completion` for `buffalo completion bash` instead of `bash`
+		cmdName = cmd.Parent().Name()
+	}
+
 	for _, freeCmd := range anywhereCommands {
-		if freeCmd != cmd.Name() {
+		if freeCmd == cmdName {
```

Comment:
If the current command (subcommand) is one of `anywhereCommands`, just returns nil and stops prechecking. Otherwise, the command should be executed within the application directory that can be determined by checking `.buffalo.dev.yml` currently.